src/lib/AirspeedValidator/AirspeedValidator.cpp

```diff
@@ -0,0 +1,277 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file AirspeedValidator.cpp
+ * Estimates airspeed scale error (from indicated to equivalent airspeed), performes
+ * checks on airspeed measurement input and reports airspeed valid or invalid.
+ */
+
+#include "AirspeedValidator.hpp"
+
+AirspeedValidator::AirspeedValidator()
+{
+}
+
+AirspeedValidator::~AirspeedValidator()
+{
+}
+
+
+void
+AirspeedValidator::update_airspeed_validator(struct airspeed_validator_update_data &input_data)
+{
+	// get indicated airspeed from input data (raw airspeed)
+	_IAS = input_data.airspeed_indicated_raw;
+
+	// to be able to detect missing data, save timestamp (used in data_missing check)
+	if (input_data.airspeed_timestamp != _previous_airspeed_timestamp && input_data.airspeed_timestamp > 0) {
+		_time_last_airspeed = input_data.timestamp;
+		_previous_airspeed_timestamp = input_data.airspeed_timestamp;
+	}
+
+	update_wind_estimator_params();
+	update_EAS_scale();
+	update_EAS_TAS(input_data.air_pressure_pa, input_data.air_temperature_celsius);
+	update_wind_estimator(input_data.timestamp, input_data.airspeed_true_raw, input_data.lpos_valid, input_data.lpos_vx,
+			      input_data.lpos_vy,
+			      input_data.lpos_vz, input_data.lpos_evh, input_data.lpos_evv, input_data.att_q);
+	update_in_fixed_wing_flight(input_data.in_fixed_wing_flight);
+	check_airspeed_innovation(input_data.timestamp, input_data.vel_test_ratio, input_data.mag_test_ratio);
+	check_load_factor(input_data.accel_z);
+	update_airspeed_valid_status(input_data.timestamp);
+}
+
+void
+AirspeedValidator::update_wind_estimator_params()
+{
+	// update wind & airspeed scale estimator parameters
+	_wind_estimator.set_wind_p_noise(_wind_estimator_wind_p_sigma);
```

Comment:
Why do you call this on each loop iteration instead of only when it's set?

Reply:
Fixed.